src/interfaces/ICoWSwapOnchainOrders.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8;
+
+import "../vendored/GPv2Order.sol";
+
+/// @title CoW Swap Onchain Order Creator Interface
+/// @author CoW Swap Developers
+interface ICoWSwapOnchainOrders {
+    /// @dev List of signature schemes that are supported by this contract to create orders onchain.
+    enum OnchainSigningScheme {
+        Eip1271,
+        PreSign
+    }
+
+    /// @dev Struct containing information on the signign scheme used plus the corresponding signature.
+    struct OnchainSignature {
+        /// @dev The signing scheme used by the signature data.
+        OnchainSigningScheme scheme;
+        /// @dev The data used as an order signature.
+        bytes data;
+    }
+
+    /// @dev Event emitted to broadcast an order onchain.
+    ///
+    /// @param sender The user who triggered the creation of the order. Note that this address does *not* need to be
+    /// the actual owner of the order and does not need to be related to the order or signature in any way.
+    /// For example, if a smart contract creates orders on behalf of the user, then the sender would be the user who
+    /// triggers the creation of the order, while the actual owner of the order would be the smart contract that
```

Comment:
>Note that it is always possible to recover the owner of the order from a valid signature.
  
Is this also true for presign orders? I was thinking to get the owner of presign orders from the backend via parsing the sender..

Reply:
This is also true for presign orders as the signature is just the signer address ([source](https://github.com/cowprotocol/contracts/blob/31b86ed0a882e669a3d6e2301bb432386204ecc5/src/contracts/mixins/GPv2Signing.sol#L304-L333)).

Something I considered instead is having two different events for presign and eip1271.. Then we don't need to have the extra `OnchainSigningScheme scheme;` field in the signature and would make everything slightly cheaper. In this case, we could also replace the signature with the owner in the presign case.

Reply:
>Something I considered instead is having two different events for presign and eip1271.

I think I am fine with both ways. For the backend, having the enum signing scheme is definitively easier, as then I only need to implement the listening for one event and make match decisions only on the signing scheme. Making the match decision on the event type is a little bit more effort, but doable as well